lua/lvim/lsp/init.lua

```diff
@@ -61,20 +61,37 @@ function M.common_on_init(client, bufnr)
 end
 
 function M.common_on_attach(client, bufnr)
+  local MB = 1024 * 1024
+  local ok, stats = pcall(vim.loop.fs_stat, vim.api.nvim_buf_get_name(bufnr))
+  local fsize = ok and stats and stats.size or 0
```

Comment:
you should be able to read the cached buffer variable already, so no need to do the calculations again.

Reply:
it later checks if the file is bigger than 5 to completely disable lsp, unless you want me to just do that on 1MB also

Reply:
Yeah, I'm thinking of using a global limit, maybe 2MB.

I'm leaning more and more towards making this a standalone plugin, with some configurations (including the limit). It might be a bit tricky to generalize it, but I think it would end up being more versatile. What do you think?